### PR TITLE
docs: adds useContext docs

### LINKED ISSUE #2953: docs: add docs for `trpc.useContext()`
### Area of Improvement

- We should add docs for `useContext`
- Invalidate Queries should probably be a sub-section for `useContext()`

### Link to related docs

https://trpc.io/docs/invalidateQueries

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing the suggested changes!

### PR DESCRIPTION
**Pull request by rhenriquez28**
Closes #2953

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Review by KATT — DISMISSED** [review]
Got a deployment error but love it apart from that.

You should be able to repro with `cd www && p build`

File: www/docs/reactjs/useContext.mdx [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+import { createTRPCReact } from '@trpc/react';
+import type { AppRouter } from './server';
+
+const trpc = createTRPCReact<AppRouter>();
+
+// ---cut---
+// @noErrors
+function MyComponent() {
+  const utils = trpc.useContext();
+  utils.post.all.f;
```
[the remark this answers is not in the document]
Comment (rhenriquez28) [reply www/docs/reactjs/useContext.mdx:58]: So I fixed the broken link in the migration guide but pnpm build doesn't give me any more errors. should I remove the twoslash sections from these then?

**Comment by rhenriquez28** [comment]
Also, in the `useInfiniteQuery` docs, there is a section where they talk about [their specific helpers](https://trpc.io/docs/v10/useInfiniteQuery#helpers). However, these come before the `useContext` docs in the sidebar. I planned to move that section to the useContext docs but I don't know if that would be the best move given that it would leave the useInfiniteQuery docs very simple. What could be the best move here?

File: www/docs/reactjs/useContext.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply www/docs/reactjs/useContext.mdx:58]: I think its fixed

**Review by juliusmarminge — DISMISSED** [review]
Lovely

**Review by bautistaaa — DISMISSED** [review]
amazing!

File: www/vercel.json [hunk 2]
```diff
@@ -54,6 +54,11 @@
       "source": "/docs/v10/publicProcedures",
       "destination": "/docs/v10/procedures",
       "permanent": true
+    },
+    {
+      "source": "/docs/v10/query-invalidation",
+      "destination": "/docs/v10/useContext#query-invalidation",
```
[the remark this answers is not in the document]
Comment (KATT) [reply www/vercel.json:60]:
```suggestion
      "destination": "/docs/v10/useContext%23query-invalidation",
```

File: www/vercel.json [hunk 3]
```diff
@@ -54,6 +54,11 @@
       "source": "/docs/v10/publicProcedures",
       "destination": "/docs/v10/procedures",
       "permanent": true
+    },
+    {
+      "source": "/docs/v10/query-invalidation",
+      "destination": "/docs/v10/useContext%23query-invalidation",
```
Comment (KATT) [inline www/vercel.json:60]:
```suggestion
      "destination": "/docs/v10/useContext#query-invalidation",
```

**Review by KATT — APPROVED** [review]

**Pull request merged**